### DISCUSSION TITLE
ESPTOOL installazione !!!RISOLTO!!!

### DISCUSSION THREAD

**Discussion by Novellin0** [indomus/forum#1599] [HUB: Homebridge & Apple HomeKit]
CIao a tutti, sono nuovo del forum e sono anche agli inizi sulla programmazione di prese smart. Stavo appunto seguendo le indicazioni per installare ESPTOOL con un Mac seguendo le indicazioni riportate su questo articolo https://indomus.it/componenti/esptool/ e nonostante abbia fatto tutti i passaggi riscontro un errore quando eseguo il comando `sudo python setup.py install`, il terminale mi dice `comando python: non trovato` eppure ho fatto tutto alla lettera (scaricato e installato Python, eseguiti tutti i comandi da terminale). Qualcuno mi sa dire dove sbaglio? Grazie

**Comment by Alexanco8** [comment]
stesso problema con mac OSX Bigsur funzionava con monterey non funziona più nemmeno facendo una nuova installazione (che mi da il tuo stesso errore)

**Reply by snakuzzo** [reply]
ma se ti dice `python` non trovato, vuol dire che il binario (o symlink) `python` non è in alcun `PATH` dell'utente con cui stai eseguendo il comando. Se eseguite un comando con `sudo` lo state di fatto eseguendo con l'utente `root` (che sia un macOS o un qualsiasi altro PC con una distro Linux)
Provate innanzi tutto a verificare con il "VOSTRO" utente dove si trova `python` o `python3` (se il symlink `python` non è stato creato):

```
which python
```

oppure

```
which python3
```

Se non li trovate, come detto sopra, vuol dire che non avete la dir in cui sono presenti i binari nella variabile `PATH`.
Se li trovate potete tranquillamente eseguire il comando...

```
python3 -m pip install esptool
```

(A patto che sia già installato `pip`)

**Reply by Alexanco8** [reply]
se faccio which python3 mi dice che python è qui /Library/Frameworks/Python.framework/Versions/3.9/bin/python3
lancio python3 -m pip install esptool e lo installa (ma ero riuscito a installarlo anche prima cambiando, rispetto alla guida indomus, python con python3)

 Il problema è nel momento i cui vado a dare il comando per flashare lo shelly: esptool.py -p /dev/tty.usbserial-0001 --baud 115200 flash_id qui mi restituisce ancora -bash: esptool.py: command not found  eppure nella cartella di esptool il file esptool.py c'è

**Reply by snakuzzo** [reply]
Prima ti ho scritto perché

**Reply by Alexanco8** [reply]
Scusa l’ignoranza quindi devo mettere la cartella di esptool in quella library (sotto utente) dove c’è python, è corretto?

**Reply by snakuzzo** [reply]
Non ho detto questo.
Prova a lanciare il comando con `sudo` con il path assoluto

**Reply by Novellin0** [reply]
RISOLTO, grazie ai consigli di @snakuzzo sono riuscito a configurare lo shelly 1

**Reply by Alexanco8** [reply]
ciao mi sto perdendo in un bicchier d'acqua probabilmente mi dici che comandi hai usato per lo shelly? io uso questo esptool.py -p /dev/tty.usbserial-0001 --baud 115200 flash_id, ho provato anche a usarlo con sudo ma niente mi dice comunque command not found e non posso procedere con gli altri comandi

**Reply by Novellin0** [reply]
ho fatto tutti i passaggi indicati e quando ho dato il comando `esptool.py -p /dev/tty.usbserial-0001 --baud 115200 flash_id` ha funzionato senza problemi. Hai verificato, con i comando `ls /dev/tty*` che il tuo device si chiami proprio così? prova a rifare tutti i passaggi dall'inizio, a me ha funzionato

**Reply by snakuzzo** [reply]
Il problema di @Alexanco8 è diverso...non è relativo alla porta. Lui non trova proprio il comando perché per root la variabile `PATH` non include il path in cui è presente il binario (o il symlink) di python.
Lancia il comando senza `sudo`.
Se invece hai necessità di lanciarlo come `root` (ma mi pare strano), allora devi usare `sudo` e, vista la tua situazione, lanciare il comando passandogli l'intero percorso (es. `sudo /Library/Frameworks/Python.framework/Versions/3.9/bin/python3...`

**Reply by Alexanco8** [reply]
la costa strana è che sia python sia esptool.py sono nella stessa path li ho appena interrogati

/Library/Frameworks/Python.framework/Versions/3.9/bin/python3
/Library/Frameworks/Python.framework/Versions/3.9/bin/esptool

la porta dell'usb l'ho interrogata ed è corretta /dev/tty.usbserial-0001
poi vado a lanciare il comando esptool.py -p /dev/tty.usbserial-0001 --baud 115200 flash_id ma anche esptool.py -p /dev/tty.usbserial-0001 erase_flash anche se provo sudo esptool.py -p /dev/tty.usbserial-0001 erase_flash ottengo sempre risposta che -bash: esptool.py: command not found

anche se faccio cd esptool e lo lancio da li non funziona, sto diventando pazzo

**Reply by snakuzzo** [reply]
Niente...quando ti ho detto di usare il path assoluto? Hai letto?

Hai provato...?

```
sudo /Library/Frameworks/Python.framework/Versions/3.9/bin/esptool.py
```

**Reply by Alexanco8** [reply]
ciao si scusami avevo provato, mi da questo risultato

sudo: /Library/Frameworks/Python.framework/Versions/3.9/bin/esptool.py: command not found

**Reply by snakuzzo** [reply]
```
sudo /Library/Frameworks/Python.framework/Versions/3.9/bin/python3 esptool.py
```

**Reply by Alexanco8** [reply]
/Library/Frameworks/Python.framework/Versions/3.9/bin/python3: can't open file '/Users/xxxxxx/esptool.py': [Errno 2] No such file or directory

**Reply by snakuzzo** [reply]
quindi esptool.py non è lì dove hai detto prima.
Ma scusa...hai provato a lanciare il comando senza `sudo` ?

se dal tuo utente fai...

```
which esptool.py
```

che ti dice ?

**Reply by snakuzzo** [reply]
Comunque NON usare sudo. Se hai installato esptool con il tuo utente devi lanciarlo con il tuo utente... fine

**Reply by Alexanco8** [reply]
Come avevo scritto prima interrogando esptool mi da questo che è lo stesso path di python è questo che non capisco sono entrambi installati col mio utente!

/Library/Frameworks/Python.framework/Versions/3.9/bin/esptool

**Reply by snakuzzo** [reply]
ripeto...poi direi che siamo andati avanti fin troppo...

Se hai installato esptool con il tuo utente devi lanciarlo con il tuo utente

**Reply by Alexanco8** [reply]
Sono nel terminale col mio utente tant’è che quando lancio il comando mi chiede la password poi però i risultati sono quelli esposti, ho utilizzato tutti i comandi che mi hai dato senza successo , grazie comunque

**Reply by snakuzzo** [reply]
Non ho più un mac per seguirti...mi spiace

**Reply by Novellin0** [reply]
> 

prova a rifare i passaggi da 0 .... secondo me o ne hai saltato qualcuno 🤷🏼‍♂️

**Reply by Alexanco8** [reply]
Che sistema operativo hai usato?

**Reply by Novellin0** [reply]
ho usato MacOS 12.3.1

**Reply by Alexanco8** [reply]
Purtroppo è un problema di mac osx monterey ho trovato altre persone nella mia situazione, mi toccherà farlo da un pc

**Reply by Novellin0** [reply]
Ma la versione 12.3.1 è Monterey

**Reply by snakuzzo** [reply]
> Purtroppo è un problema di mac osx monterey ho trovato altre persone nella mia situazione, mi toccherà farlo da un pc

il problema non credo proprio sia macOS...

**Reply by Novellin0** [reply]
> > Purtroppo è un problema di mac osx monterey ho trovato altre persone nella mia situazione, mi toccherà farlo da un pc
> 
> il problema non credo proprio sia macOS...

infatti 🤷🏼‍♂️

**Reply by snakuzzo** [reply]
Comunuqe...se con il tuo utente riesci ad usare `python3` prova ad installarlo così (con il tuo utente...senza usare `sudo`)...

```
python3 -m pip install esptool --upgrade
```

(il `--upgrade` serve nel caso in cui tu l'abbia già installato e voglia aggiornarlo)

**Reply by snakuzzo** [reply]
Fatto ora su linux...la sostanza è la stessa

```
$ python3 -m pip install esptool --upgrade
Requirement already satisfied: esptool in ./.local/lib/python3.9/site-packages (3.2)
Collecting esptool
  Downloading esptool-3.3.tar.gz (213 kB)
     |████████████████████████████████| 213 kB 2.7 MB/s 
Requirement already satisfied: bitstring>=3.1.6 in ./.local/lib/python3.9/site-packages (from esptool) (3.1.9)
Requirement already satisfied: cryptography>=2.1.4 in /usr/lib/python3/dist-packages (from esptool) (3.3.2)
Requirement already satisfied: ecdsa>=0.16.0 in ./.local/lib/python3.9/site-packages (from esptool) (0.17.0)
Requirement already satisfied: pyserial>=3.0 in ./.local/lib/python3.9/site-packages (from esptool) (3.5)
Requirement already satisfied: reedsolo<=1.5.4,>=1.5.3 in ./.local/lib/python3.9/site-packages (from esptool) (1.5.4)
Requirement already satisfied: six>=1.9.0 in /usr/lib/python3/dist-packages (from ecdsa>=0.16.0->esptool) (1.16.0)
Building wheels for collected packages: esptool
  Building wheel for esptool (setup.py) ... done
  Created wheel for esptool: filename=esptool-3.3-py3-none-any.whl size=350602 sha256=55eae3e5ec2cbff746dc57c7b32db8853b2cb6ba81ee955f271d6365cda1c512
  Stored in directory: /home/snakuzzo/.cache/pip/wheels/b9/bf/88/fa6ce266036e142d5cf1d1b6b9cc1c1b7a1b0fcc5ebb29b0b9
Successfully built esptool
Installing collected packages: esptool
  Attempting uninstall: esptool
    Found existing installation: esptool 3.2
    Uninstalling esptool-3.2:
      Successfully uninstalled esptool-3.2
Successfully installed esptool-3.3
$
$
$ esptool.py version
esptool.py v3.3
3.3
```

**Reply by Alexanco8** [reply]
> Ma la versione 12.3.1 è Monterey

Hai scritto ios pensavo pertanto non ti riferissi a osx

**Reply by Novellin0** [reply]
> > Ma la versione 12.3.1 è Monterey
> 
> Hai scritto ios pensavo pertanto non ti riferissi a osx

Scusa, avevo sbagliato poi ho ricorretto 🤗

**Reply by Alexanco8** [reply]
Figurati comunque ribadisco che quando avevo bigsur ho installato ed usato senza alcun problema esptool e python pertanto la procedura la conosco, improvvisamente ha smesso di funzionare ho rifatto la procedura su due mac distinti entrambi con monterey e l’esito è questo

**Reply by Novellin0** [reply]
> Figurati comunque ribadisco che quando avevo bigsur ho installato ed usato senza alcun problema esptool e python pertanto la procedura la conosco, improvvisamente ha smesso di funzionare ho rifatto la procedura su due mac distinti entrambi con monterey e l’esito è questo

che vuoi che ti dica, seguendo le indicazioni di @snakuzzo sono riuscito a concludere la procedura, ma nel tuo caso sembra come se non riconosca il comando `esptool.py` 🤔

**Reply by snakuzzo** [reply]
Scrivi `esptool.py`... non `esptool`
